src/plugins/ui_core_views/cell_evaluation/evaluator.ts

```diff
@@ -93,11 +93,9 @@ export class Evaluator {
   }
 
   getArrayFormulaSpreadingOn(position: CellPosition): CellPosition | undefined {
-    const hasArrayFormulaResult =
-      this.getEvaluatedCell(position).type !== CellValueType.empty &&
-      !this.getters.getCell(position)?.isFormula;
-    if (!hasArrayFormulaResult) {
-      return this.spreadingRelations.isArrayFormula(position) ? position : undefined;
+    const hasNoResult = this.getEvaluatedCell(position).type === CellValueType.empty;
```

Comment:
nitpick: slightly easier to read when there no negation
```suggestion
    const isEmpty = this.getEvaluatedCell(position).type === CellValueType.empty;
```